javacutil/src/main/java/org/checkerframework/javacutil/Pair.java

```diff
@@ -22,6 +22,70 @@ public static <V1, V2> Pair<V1, V2> of(V1 v1, V2 v2) {
     return new Pair<>(v1, v2);
   }
 
+  /**
+   * Returns a copy of this in which each element is a clone of the corresponding element of this.
+   * {@code clone()} may or may not itself make a deep copy of the elements.
+   *
+   * @param <V1> the type of the first element of the pair
+   * @param <V2> the type of the second element of the pair
+   * @param orig a pair
+   * @return a copy of {@code orig}, with all elements cloned
+   */
+  @SuppressWarnings("nullness") // generics problem with deepCopy()
+  public static <V1 extends Cloneable, V2 extends Cloneable> Pair<V1, V2> cloneElements(
+      Pair<V1, V2> orig) {
+    // Cannot modify result of super.clone() because fields are final.
+
+    V1 oldFirst = orig.first;
+    V1 newFirst = oldFirst == null ? oldFirst : CollectionUtils.clone(oldFirst);
+    V2 oldSecond = orig.second;
+    V2 newSecond = oldSecond == null ? oldSecond : CollectionUtils.clone(oldSecond);
+    return of(newFirst, newSecond);
+  }
+
+  /**
+   * Returns a deep copy of this: each element is a deep copy of the corresponding element of this.
+   *
+   * @param <V1> the type of the first element of the pair
+   * @param <V2> the type of the second element of the pair
+   * @param orig a pair
+   * @return a deep copy of {@code orig}
+   */
+  @SuppressWarnings("nullness") // generics problem with deepCopy()
+  public static <V1 extends DeepCopyable, V2 extends DeepCopyable> Pair<V1, V2> deepCopy(
+      Pair<V1, V2> orig) {
+    // Cannot modify result of super.clone() because fields are final.
```

Comment:
`clone()` isn't called in this method (anymore?), so I think this comment should either be deleted or changed to refer to `deepCopy()`